tools/integration_tests/kernel-list-cache/infinite_kernel_list_cache_test.go

```diff
@@ -88,6 +88,152 @@ func (s *infiniteKernelListCacheTest) TestKernelListCache_AlwaysCacheHit(t *test
 	assert.Equal(t, "file2.txt", names2[1])
 }
 
+// (a) First ReadDir() will be served from GCSFuse filesystem.
+// (b) Second ReadDir() will also be served from GCSFuse filesystem, because of
+// addition of new file.
+func (s *infiniteKernelListCacheTest) TestKernelListCache_CacheMissOnAdditionOfFile(t *testing.T) {
+	targetDir := path.Join(testDirPath, "explicit_dir")
```

Comment:
If this is something we have to do for every test, shall we add it to setup function?

Reply:
Isn't it in guideline of writing tests that we should not sacrifice readability over  repetition?

Reply:
I don't think moving the directory setup to setup function will reduce test readability since it is part of the setup but if you think otherwise, you can skip :)

Reply:
then targetDir and all we have to keep it as global variable. Prince was not align with it. So let's keep it like this only.